Classes/Steno.sc

```diff
@@ -469,20 +474,47 @@ Steno {
 
 		// same for all closing brackets
 		// mix controls balance between serial bus result and outside bus
-		routingFunction = { |in, out, dryIn, mix = 1, through = 0| // mix = 1: don't add outside in twice
-			var input = In.ar(in, numChannels); // in: result of serial synths
-			var oldSignal = In.ar(out, numChannels); // the old signal on the bus, mixed in by through
-			var inputOutside = In.ar(dryIn, numChannels);  // dryIn: bus outside parenthesis
-			var signalOnMixBus = input + (through * oldSignal);
-			var output = XFade2.ar(inputOutside, signalOnMixBus, mix * 2 - 1);
-			ReplaceOut.ar(in, Silent.ar(numChannels)); // clean up: overwrite channel with zero.
-			XOut.ar(out, EnvGate.new, output); // overwrite the out channel with the new mix
+		routingFunction = {// |in, out, dryIn, mix = 1, through = 0| // mix = 1: don't add outside in twice
+			var oldSignal, inputOutside, stenoSignal, signal, signalOnMixBus;
+
+			stenoSignal = StenoSignal(numChannels);
+			stenoSignal.filterInput;
+			oldSignal = In.ar(stenoSignal.outBus, numChannels); // the old signal on the bus, mixed in by through
+			inputOutside = In.ar(stenoSignal.dryIn, numChannels);  // dryIn: bus outside parenthesis
+
+			signal = XFade2.ar(inputOutside, stenoSignal.input, MulAdd(stenoSignal.mix, 2, -1));
+
+			signal = Mix.ar([
+				signal,
+				oldSignal * max(
+					stenoSignal.through,
+					1 - stenoSignal.env
+				)
+			]);   // fade old input according to gate, signal is supposed to fade out itself.
+
+			FreeSelfWhenDone.kr(stenoSignal.env); // free synth if gate 0
+			ReplaceOut.ar(stenoSignal.inBus, Silent.ar(numChannels)); // clean up: overwrite channel with zero.
+
+			stenoSignal.addOutput(signal);
+			stenoSignal.writeToBus;
 		};
 
 		// nothing to do, just clean up bus, to be sure.
-		dummyOpeningFunction = { |in, out|
-			ReplaceOut.ar(out, Silent.ar(numChannels)); // umbrella
-			FreeSelf.kr(\gate.kr(1) < 1); // dummy synth, can be released
+		// dummyOpeningFunction = {
+		// 	var stenoSignal;
+
+		// 	stenoSignal = StenoSignal(numChannels);
+		// 	stenoSignal.filterInput;
+		// 	ReplaceOut.ar(stenoSignal.outBus, Silent.ar(numChannels)); // umbrella
+		// 	FreeSelfWhenDone.kr(stenoSignal.env);
+		// };
+		dummyOpeningFunction = {
+			var stenoSignal;
+			stenoSignal = StenoSignal(numChannels);
+			stenoSignal.quelle(nil, true, numChannels);
```

Comment:
`stenoSignal.quelle(nil, …)` will do nothing but `stenoSignal.quelleInput`, which in turn does nothing here.

Why is it here?

Reply:
good question. will look at it.